common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -643,17 +637,58 @@ def get_problem_html(self, encapsulate=True):
         demand_hints = self.lcp.tree.xpath("//problem/demandhint/hint")
         demand_hint_possible = len(demand_hints) > 0
 
+        # Get the current problem status and generate the answer notification and message.
+        answer_notification_message = None
+        answer_notification_type = None
+        progress = self.get_progress()
+
+        if progress is not None:
+            key_list = self.correct_map.keys()
+            if len(key_list) == 1:
+                # Only one answer available or a missing progress and 1 or more answers.
+                answer_notification_type = self.correct_map.get(key_list[0]).get('correctness', None)
+            elif len(key_list) > 1:
+                # Look at the progress as we only want to show one notification for the problem. In this case there
+                # can be more that one correctness state in the correct_map.
+                if 0.0 < progress.percent() < 100.0:
```

Comment:
And isn't there existing logic someplace else to determine this partially correct/correct/incorrect state? In prod today we do show different icons, correct?